tajo-core/src/main/java/org/apache/tajo/master/rm/TajoWorkerResourceManager.java

```diff
@@ -221,11 +231,16 @@ public WorkerAllocatedResource allocateQueryMaster(QueryInProgress queryInProgre
     try {
       response = callFuture.get(3, TimeUnit.SECONDS);
     } catch (Throwable t) {
-      LOG.error(t, t);
-      return null;
+      response = callFuture.cancel(); // try cancel
+      if (response == null) {
+        // canceled future
+        LOG.warn("Got exception " + t);
+        LOG.error(t, t);
```

Comment:
I think this line is not necessary if you change as the following in the immediately preceding line.
```
LOG.warn("Got exception while allocating QueryMaster: " + t, t);  
```